### ISSUE TITLE
[BUG]: Llama2 HybridParallelPlugin train failed when  pp_size>1

### ISSUE THREAD

**Issue by xs1997zju** [hpcaitech/ColossalAI#4705]
### 🐛 Describe the bug

Llama2 HybridParallelPlugin train failed when  pp_size>1
we modified the llama training example from the llama model to the falcon model but encountered such an error.
The hidden_states is None in  colossalai/shardformer/modeling/llama.py:61 in  llama_model_forward
<img width="739" alt="image" src="https://github.com/hpcaitech/ColossalAI/assets/106210907/7deed690-6f8b-462c-9d05-1293a3d2eae5">

### Environment

torch-'1.13.1+cu117', transformers-4.32.0, colossalAI-release-v0.3.2

**Comment by flybird11111** [comment]
Hi, The example of using the hybrid parallel plugin in llama2 is still in development and has not been merged. If you encounter any issues after making modifications, you can join our WeChat group, and we can help you take a look.

**Comment by flybird11111** [comment]
> Thanks, but in colossalAI PRs, you put SharFormer example of llama, and says support llama1/2. This exaggerates the functionality of the codebase, causing confusion and increasing debugging costs for users. We hope you can control the level of exaggeration in your PRs. Your project is indeed excellent, but the promotion is too excessive. <img alt="image" width="994" src="https://user-images.githubusercontent.com/106210907/267573103-845eecb4-5692-4ada-82bc-3196c0672a8e.png">

LlamaV2 is also supported.

**Comment by xs1997zju** [comment]
?? I just report the bug of useing pp_size=2 in the PR's example. Does we discuss the same thing?

**Comment by Fridge003** [comment]
> ### 🐛 Describe the bug
> Llama2 HybridParallelPlugin train failed when pp_size>1 we modified the llama training example to use hybridparallel plugin but encountered such an error. The hidden_states is None in colossalai/shardformer/modeling/llama.py:61 in llama_model_forward <img alt="image" width="739" src="https://user-images.githubusercontent.com/106210907/267564835-7deed690-6f8b-462c-9d05-1293a3d2eae5.png">
> 
> ### Environment
> torch-'1.13.1+cu117', transformers-4.32.0, colossalAI-release-v0.3.2

Hi, this is a typical bug. When enabling pipeline parallel, please do not do forward/backward through model(input)/loss.backward(), rather, please call `booster.execute_pipeline` to do forward/backward. For concrete usage you can refer to [this example](https://github.com/hpcaitech/ColossalAI/blob/main/examples/language/bert/finetune.py). 

The example of LlaMa2 is still under development, we apologize for that, and it will be released within a few days. Thank you for your issue !

**Comment by flybird11111** [comment]
> ?? I just report the bug of useing pp_size=2 in the PR's example. Does we discuss the same thing?

I understand, and we will keep an eye on this issue.

**Comment by xs1997zju** [comment]
Okay, thank you  for the prompt response, appreciate it.
